application-collabora-default/src/main/java/com/xwiki/collabora/internal/configuration/AbstractCollaboraConfigurationSource.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.collabora.internal.configuration;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.xwiki.configuration.internal.AbstractDocumentConfigurationSource;
+import org.xwiki.model.reference.LocalDocumentReference;
+
+/**
+ * Common features for all Collabora configuration sources.
```

Comment:
Would be good to add a TODO comment to remove this class once we upgrade the parent to XWiki 14.9+ where we have the ``AbstractWikisConfigurationSource`` that does this fallback on the main wiki automatically. See https://github.com/xwiki/xwiki-platform/commit/2a9f7e3bed025fec4afb77713ba9aad9e55cf129 and https://jira.xwiki.org/browse/XWIKI-20195 .